{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3b807a5f-c783-4225-993f-3fd1ae2cdfbe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘../06_phylos/analysis-iqtree’: File exists\n",
      "IQ-TREE multicore version 2.4.0 for Linux x86 64-bit built Feb 12 2025\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
      "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
      "\n",
      "Host:    cantley (AVX512, FMA3, 377 GB RAM)\n",
      "Command: iqtree2 -s ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy -m MFP --alrt 10000 -B 10000 -T 10 -redo\n",
      "Seed:    918298 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Aug  4 00:01:06 2025\n",
      "Kernel:  AVX+FMA - 10 threads (10 CPU cores detected)\n",
      "\n",
      "Reading alignment file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy ... Phylip format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 95 sequences with 18736 columns, 17602 distinct patterns\n",
      "7367 parsimony-informative, 5056 singleton sites, 6313 constant sites\n",
      "          Gap/Ambiguity  Composition  p-value\n",
      "Analyzing sequences: done in 0.00194751 secs using 1016% CPU\n",
      "   1  CO-AR-1    11.40%    passed     25.05%\n",
      "   2  CO-AR-10   10.13%    passed     23.38%\n",
      "   3  CO-AR-11   10.77%    passed     31.92%\n",
      "   4  CO-AR-12    6.19%    passed     55.04%\n",
      "   5  CO-AR-13   10.65%    passed     34.34%\n",
      "   6  CO-AR-14    7.00%    passed     72.24%\n",
      "   7  CO-AR-15    6.94%    passed     60.20%\n",
      "   8  CO-AR-16    6.28%    passed     63.68%\n",
      "   9  CO-AR-17   10.27%    passed     51.63%\n",
      "  10  CO-AR-18    7.29%    passed     39.52%\n",
      "  11  CO-AR-19   10.69%    passed     31.05%\n",
      "  12  CO-AR-2    11.01%    passed     19.02%\n",
      "  13  CO-AR-20    6.98%    passed     64.77%\n",
      "  14  CO-AR-3    11.31%    passed     19.78%\n",
      "  15  CO-AR-4    10.94%    passed     68.74%\n",
      "  16  CO-AR-5    11.41%    passed     34.68%\n",
      "  17  CO-AR-6    10.67%    passed     18.58%\n",
      "  18  CO-AR-7    11.34%    passed     28.77%\n",
      "  19  CO-AR-8     8.58%    passed     42.14%\n",
      "  20  CO-AR-9     5.38%    passed     73.46%\n",
      "  21  CO-CH-11   12.95%    passed     23.99%\n",
      "  22  CO-CH-12   12.95%    passed     79.68%\n",
      "  23  CO-CH-2     7.50%    passed     77.51%\n",
      "  24  CO-CH-21    6.18%    passed     98.27%\n",
      "  25  CO-CH-23   17.42%    passed     82.16%\n",
      "  26  CO-CH-39   10.09%    passed     93.92%\n",
      "  27  CO-CH-40    9.72%    passed     91.43%\n",
      "  28  CO-CH-45    7.27%    passed     81.52%\n",
      "  29  CO-CH-46   12.75%    passed     34.28%\n",
      "  30  CO-CH-5     6.59%    passed     99.14%\n",
      "  31  CO-CH-6    10.00%    passed     65.52%\n",
      "  32  CO-CH-8    11.67%    passed     43.11%\n",
      "  33  CO-CH-9    10.51%    passed     51.76%\n",
      "  34  CO-IP-1    10.17%    passed     32.12%\n",
      "  35  CO-IP-10   11.91%    failed      4.86%\n",
      "  36  CO-IP-2    10.19%    passed     42.62%\n",
      "  37  CO-IP-3    11.44%    passed     27.04%\n",
      "  38  CO-IP-4    11.64%    passed     31.41%\n",
      "  39  CO-IP-5    11.60%    passed     20.48%\n",
      "  40  CO-IP-6     7.33%    passed     75.40%\n",
      "  41  CO-IP-7    11.12%    passed     33.14%\n",
      "  42  CO-IP-8     7.26%    passed     93.69%\n",
      "  43  CO-IP-9    11.60%    passed     28.11%\n",
      "  44  CO-MO-1     6.45%    passed     58.69%\n",
      "  45  CO-MO-10   10.99%    passed     62.02%\n",
      "  46  CO-MO-11   13.18%    passed     30.06%\n",
      "  47  CO-MO-13    4.24%    passed     97.76%\n",
      "  48  CO-MO-14    8.39%    passed     41.43%\n",
      "  49  CO-MO-15    8.28%    passed     78.97%\n",
      "  50  CO-MO-16    9.91%    passed     48.00%\n",
      "  51  CO-MO-17   12.17%    passed     32.68%\n",
      "  52  CO-MO-18    9.04%    passed     58.34%\n",
      "  53  CO-MO-19   11.83%    passed     47.40%\n",
      "  54  CO-MO-20    8.89%    passed     65.07%\n",
      "  55  CO-MO-3     6.37%    passed     91.78%\n",
      "  56  CO-MO-4     4.89%    passed     92.58%\n",
      "  57  CO-MO-5    10.61%    passed     48.46%\n",
      "  58  CO-MO-6    12.77%    passed     29.06%\n",
      "  59  CO-MO-7    12.27%    passed      9.38%\n",
      "  60  CO-MO-8     8.64%    passed     86.28%\n",
      "  61  CO-MO-9    12.15%    passed     47.73%\n",
      "  62  CO-PB-1    13.16%    passed     47.48%\n",
      "  63  CO-PB-10   13.63%    passed     54.32%\n",
      "  64  CO-PB-11   13.10%    passed     40.48%\n",
      "  65  CO-PB-12   13.16%    passed     50.53%\n",
      "  66  CO-PB-13   12.78%    passed     43.26%\n",
      "  67  CO-PB-14   13.72%    passed     40.78%\n",
      "  68  CO-PB-15   13.44%    passed     44.71%\n",
      "  69  CO-PB-16   17.02%    passed     81.13%\n",
      "  70  CO-PB-17   13.47%    passed     64.98%\n",
      "  71  CO-PB-18   13.18%    passed     95.67%\n",
      "  72  CO-PB-19   13.41%    passed     92.86%\n",
      "  73  CO-PB-2    13.01%    passed     49.05%\n",
      "  74  CO-PB-20   13.41%    passed     85.06%\n",
      "  75  CO-PB-3    12.34%    passed     61.76%\n",
      "  76  CO-PB-4    14.55%    passed     58.23%\n",
      "  77  CO-PB-5    13.43%    passed     60.00%\n",
      "  78  CO-PB-6    11.74%    passed     99.82%\n",
      "  79  CO-PB-7    12.06%    passed     37.88%\n",
      "  80  CO-PB-8    13.13%    passed     58.23%\n",
      "  81  CO-PB-9    12.70%    passed     54.16%\n",
      "  82  CO-PK-10   12.33%    passed     54.90%\n",
      "  83  CO-PK-11   22.14%    passed     80.26%\n",
      "  84  CO-PK-12   12.22%    passed     59.39%\n",
      "  85  CO-PK-14   12.20%    passed     51.42%\n",
      "  86  CO-PK-15   12.40%    passed     54.27%\n",
      "  87  CO-PK-16    9.03%    passed     88.86%\n",
      "  88  CO-PK-17   11.64%    passed     53.63%\n",
      "  89  CO-PK-18    8.11%    passed     84.95%\n",
      "  90  CO-PK-19   17.58%    passed     94.69%\n",
      "  91  CO-PK-20   12.18%    passed     55.98%\n",
      "  92  CO-PK-3    12.02%    passed     37.03%\n",
      "  93  CO-PK-4    17.70%    passed     68.24%\n",
      "  94  CO-PK-6    11.33%    passed     26.90%\n",
      "  95  CO-PK-9    11.51%    passed     30.18%\n",
      "****  TOTAL      11.00%  1 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.182 seconds\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -191613.561\n",
      "2. Current log-likelihood: -191608.892\n",
      "3. Current log-likelihood: -191604.901\n",
      "4. Current log-likelihood: -191601.380\n",
      "5. Current log-likelihood: -191598.245\n",
      "6. Current log-likelihood: -191595.441\n",
      "7. Current log-likelihood: -191592.914\n",
      "8. Current log-likelihood: -191590.628\n",
      "9. Current log-likelihood: -191588.548\n",
      "10. Current log-likelihood: -191586.652\n",
      "11. Current log-likelihood: -191584.916\n",
      "12. Current log-likelihood: -191583.325\n",
      "13. Current log-likelihood: -191581.861\n",
      "14. Current log-likelihood: -191580.512\n",
      "15. Current log-likelihood: -191579.266\n",
      "16. Current log-likelihood: -191578.113\n",
      "17. Current log-likelihood: -191577.046\n",
      "Optimal log-likelihood: -191576.048\n",
      "Rate parameters:  A-C: 1.05462  A-G: 2.64242  A-T: 1.77001  C-G: 0.59917  C-T: 2.63732  G-T: 1.00000\n",
      "Base frequencies:  A: 0.249  C: 0.253  G: 0.250  T: 0.248\n",
      "Proportion of invariable sites: 0.015\n",
      "Gamma shape alpha: 0.965\n",
      "Parameters optimization took 17 rounds (4.055 sec)\n",
      "Time for fast ML tree search: 8.821 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 1214 MB RAM!\n",
      "ModelFinder will test up to 484 DNA models (sample size: 18736 epsilon: 0.100) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  GTR+F         196022.365   195 392434.730   392438.853   393963.179\n",
      "  2  GTR+F+I       194385.234   196 389162.468   389166.634   390698.756\n",
      "  3  GTR+F+G4      191560.679   196 383513.359   383517.524   385049.647\n",
      "  4  GTR+F+I+G4    191562.460   197 383518.920   383523.129   385063.046\n",
      "  5  GTR+F+R2      190650.797   197 381695.594   381699.803   383239.720\n",
      "  6  GTR+F+R3      190372.097   199 381142.193   381146.488   382701.996\n",
      "  7  GTR+F+R4      190292.943   201 380987.885   380992.267   382563.364\n",
      "  8  GTR+F+R5      190292.892   203 380991.784   380996.253   382582.939\n",
      " 14  GTR+F+I+R2    190653.528   198 381703.057   381707.308   383255.021\n",
      " 15  GTR+F+I+R3    190372.151   200 381144.302   381148.639   382711.942\n",
      " 16  GTR+F+I+R4    190292.948   202 380989.896   380994.321   382573.213\n",
      " 17  GTR+F+I+R5    190292.893   204 380993.787   380998.300   382592.780\n",
      " 29  SYM+R4        190286.028   198 380968.057   380972.308   382520.021\n",
      " 38  SYM+I+R4      190286.060   199 380970.120   380974.414   382529.922\n",
      " 51  TVM+F+R4      190293.164   200 380986.329   380990.667   382553.969\n",
      " 60  TVM+F+I+R4    190293.221   201 380988.443   380992.824   382563.921\n",
      " 73  TVMe+R4       190286.042   197 380966.085   380970.293   382510.211\n",
      " 82  TVMe+I+R4     190286.110   198 380968.220   380972.471   382520.184\n",
      " 95  TIM3+F+R4     190706.479   199 381810.957   381815.252   383370.760\n",
      "104  TIM3+F+I+R4   190706.499   200 381812.998   381817.336   383380.639\n",
      "117  TIM3e+R4      190695.190   196 381782.380   381786.546   383318.668\n",
      "126  TIM3e+I+R4    190695.217   197 381784.434   381788.643   383328.560\n",
      "139  TIM2+F+R4     190643.235   199 381684.470   381688.764   383244.272\n",
      "148  TIM2+F+I+R4   190643.266   200 381686.532   381690.869   383254.172\n",
      "161  TIM2e+R4      190616.508   196 381625.017   381629.182   383161.304\n",
      "170  TIM2e+I+R4    190616.517   197 381627.034   381631.242   383171.159\n",
      "183  TIM+F+R4      191039.959   199 382477.917   382482.212   384037.719\n",
      "192  TIM+F+I+R4    191039.965   200 382479.929   382484.267   384047.570\n",
      "205  TIMe+R4       191008.906   196 382409.812   382413.977   383946.099\n",
      "214  TIMe+I+R4     191008.916   197 382411.833   382416.041   383955.959\n",
      "227  TPM3u+F+R4    190707.314   198 381810.628   381814.879   383362.592\n",
      "236  TPM3u+F+I+R4  190707.307   199 381812.614   381816.908   383372.416\n",
      "249  TPM3+R4       190695.656   195 381781.313   381785.436   383309.762\n",
      "258  TPM3+I+R4     190695.659   196 381783.317   381787.483   383319.605\n",
      "271  TPM2u+F+R4    190643.359   198 381682.718   381686.969   383234.682\n",
      "280  TPM2u+F+I+R4  190643.361   199 381684.722   381689.016   383244.524\n",
      "293  TPM2+R4       190616.477   195 381622.954   381627.077   383151.403\n",
      "302  TPM2+I+R4     190616.481   196 381624.962   381629.127   383161.250\n",
      "315  K3Pu+F+R4     191040.402   198 382476.804   382481.055   384028.768\n",
      "324  K3Pu+F+I+R4   191040.404   199 382478.809   382483.103   384038.611\n",
      "337  K3P+R4        191009.091   195 382408.182   382412.305   383936.632\n",
      "346  K3P+I+R4      191009.088   196 382410.176   382414.341   383946.463\n",
      "359  TN+F+R4       191054.494   198 382504.988   382509.239   384056.952\n",
      "368  TN+F+I+R4     191054.495   199 382506.989   382511.284   384066.791\n",
      "381  TNe+R4        191023.463   195 382436.925   382441.048   383965.375\n",
      "390  TNe+I+R4      191023.463   196 382438.925   382443.091   383975.213\n",
      "403  HKY+F+R4      191054.957   197 382503.915   382508.123   384048.040\n",
      "412  HKY+F+I+R4    191054.957   198 382505.915   382510.166   384057.879\n",
      "425  K2P+R4        191023.652   194 382435.305   382439.386   383955.916\n",
      "434  K2P+I+R4      191023.652   195 382437.305   382441.428   383965.754\n",
      "447  F81+F+R4      193803.803   196 387999.607   388003.772   389535.894\n",
      "456  F81+F+I+R4    193803.803   197 388001.605   388005.814   389545.731\n",
      "469  JC+R4         193772.217   193 387930.433   387934.472   389443.206\n",
      "478  JC+I+R4       193772.220   194 387932.440   387936.520   389453.051\n",
      "Akaike Information Criterion:           TVMe+R4\n",
      "Corrected Akaike Information Criterion: TVMe+R4\n",
      "Bayesian Information Criterion:         TVMe+R4\n",
      "Best-fit model: TVMe+R4 chosen according to BIC\n",
      "\n",
      "All model information printed to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.model.gz\n",
      "CPU time for ModelFinder: 3243.761 seconds (0h:54m:3s)\n",
      "Wall-clock time for ModelFinder: 336.970 seconds (0h:5m:36s)\n",
      "Generating 10000 samples for ultrafast bootstrap (seed: 918298)...\n",
      "\n",
      "NOTE: 888 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -193772.216\n",
      "2. Current log-likelihood: -190287.252\n",
      "3. Current log-likelihood: -190286.046\n",
      "Optimal log-likelihood: -190285.989\n",
      "Rate parameters:  A-C: 1.05771  A-G: 2.67224  A-T: 1.73085  C-G: 0.61073  C-T: 2.67224  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.813,0.506) (0.154,2.555) (0.032,5.412) (0.001,23.895)\n",
      "Parameters optimization took 3 rounds (3.719 sec)\n",
      "Wrote distance file to... \n",
      "Computing ML distances based on estimated model parameters...\n",
      "Calculating distance matrix: done in 0.0452085 secs using 1010% CPU\n",
      "Computing ML distances took 0.050584 sec (of wall-clock time) 0.485961 sec (of CPU time)\n",
      "Setting up auxiliary I and S matrices: done in 0.00460851 secs using 6.423% CPU\n",
      "Computing RapidNJ tree took 0.332458 sec (of wall-clock time) 0.271879 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -191210.120\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 22.898 second\n",
      "Computing log-likelihood of 98 initial trees ... 10.785 seconds\n",
      "Current best score: -190125.705\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -189963.023\n",
      "Iteration 10 / LogL: -189986.970 / Time: 0h:7m:4s\n",
      "Iteration 20 / LogL: -190093.091 / Time: 0h:7m:50s\n",
      "Finish initializing candidate tree set (17)\n",
      "Current best tree score: -189963.023 / CPU time: 124.466\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 26: -189958.009\n",
      "Iteration 30 / LogL: -189959.795 / Time: 0h:8m:17s (0h:8m:50s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 40: -189956.869\n",
      "Iteration 40 / LogL: -189956.869 / Time: 0h:8m:41s (0h:7m:53s left)\n",
      "Iteration 50 / LogL: -190045.818 / Time: 0h:9m:5s (0h:6m:23s left)\n",
      "Log-likelihood cutoff on original alignment: -190253.106\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 57: -189954.840\n",
      "Iteration 60 / LogL: -190203.740 / Time: 0h:9m:33s (0h:6m:29s left)\n",
      "Iteration 70 / LogL: -189961.417 / Time: 0h:9m:59s (0h:5m:30s left)\n",
      "Iteration 80 / LogL: -189961.262 / Time: 0h:10m:22s (0h:4m:38s left)\n",
      "Iteration 90 / LogL: -189986.559 / Time: 0h:10m:46s (0h:3m:52s left)\n",
      "Iteration 100 / LogL: -189957.595 / Time: 0h:11m:11s (0h:3m:12s left)\n",
      "Log-likelihood cutoff on original alignment: -190253.106\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.998\n",
      "Iteration 110 / LogL: -189966.181 / Time: 0h:11m:39s (0h:4m:59s left)\n",
      "Iteration 120 / LogL: -189955.150 / Time: 0h:12m:5s (0h:4m:20s left)\n",
      "Iteration 130 / LogL: -190040.474 / Time: 0h:12m:29s (0h:3m:43s left)\n",
      "Iteration 140 / LogL: -189962.446 / Time: 0h:12m:55s (0h:3m:9s left)\n",
      "Iteration 150 / LogL: -190015.256 / Time: 0h:13m:18s (0h:2m:35s left)\n",
      "Log-likelihood cutoff on original alignment: -190253.106\n",
      "TREE SEARCH COMPLETED AFTER 158 ITERATIONS / Time: 0h:13m:42s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -189954.840\n",
      "2. Current log-likelihood: -189954.787\n",
      "3. Current log-likelihood: -189954.752\n",
      "Optimal log-likelihood: -189954.741\n",
      "Rate parameters:  A-C: 1.05947  A-G: 2.67390  A-T: 1.73884  C-G: 0.60874  C-T: 2.67390  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.813,0.507) (0.155,2.562) (0.031,5.489) (0.001,24.424)\n",
      "Parameters optimization took 3 rounds (1.246 sec)\n",
      "BEST SCORE FOUND : -189954.741\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 10000 replicates...\n",
      "31.354 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex\n",
      "Total tree length: 1.738\n",
      "\n",
      "Total number of iterations: 158\n",
      "CPU time used for tree search: 4359.954 sec (1h:12m:39s)\n",
      "Wall-clock time used for tree search: 476.540 sec (0h:7m:56s)\n",
      "Total CPU time used: 7977.108 sec (2h:12m:57s)\n",
      "Total wall-clock time used: 857.495 sec (0h:14m:17s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex...\n",
      "95 taxa and 445 splits.\n",
      "Consensus tree written to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "Reading input trees file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "Log-likelihood of consensus tree: -189954.759\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.iqtree\n",
      "  Maximum-likelihood tree:       ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.treefile\n",
      "  Likelihood distances:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex\n",
      "  Consensus tree:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "  Screen log file:               ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss20.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.log\n",
      "\n",
      "Date and Time: Mon Aug  4 00:15:25 2025\n",
      "IQ-TREE multicore version 2.4.0 for Linux x86 64-bit built Feb 12 2025\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
      "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
      "\n",
      "Host:    cantley (AVX512, FMA3, 377 GB RAM)\n",
      "Command: iqtree2 -s ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy -m MFP --alrt 10000 -B 10000 -T 10 -redo\n",
      "Seed:    990625 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Aug  4 00:15:25 2025\n",
      "Kernel:  AVX+FMA - 10 threads (10 CPU cores detected)\n",
      "\n",
      "Reading alignment file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy ... Phylip format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 101 sequences with 19088 columns, 18960 distinct patterns\n",
      "7773 parsimony-informative, 5288 singleton sites, 6027 constant sites\n",
      "          Gap/Ambiguity  Composition  p-value\n",
      "Analyzing sequences: done in 0.00601869 secs using 587.1% CPU\n",
      "   1  CO-AR-1    11.19%    passed     23.81%\n",
      "   2  CO-AR-10    9.95%    passed     21.12%\n",
      "   3  CO-AR-11   10.60%    passed     29.90%\n",
      "   4  CO-AR-12    6.11%    passed     52.33%\n",
      "   5  CO-AR-13   10.47%    passed     32.63%\n",
      "   6  CO-AR-14    6.88%    passed     69.07%\n",
      "   7  CO-AR-15    6.83%    passed     58.61%\n",
      "   8  CO-AR-16    6.17%    passed     62.00%\n",
      "   9  CO-AR-17   10.10%    passed     47.47%\n",
      "  10  CO-AR-18    7.17%    passed     38.47%\n",
      "  11  CO-AR-19   10.51%    passed     29.33%\n",
      "  12  CO-AR-2    10.83%    passed     19.02%\n",
      "  13  CO-AR-20    6.86%    passed     62.10%\n",
      "  14  CO-AR-3    11.10%    passed     18.69%\n",
      "  15  CO-AR-4    10.72%    passed     70.35%\n",
      "  16  CO-AR-5    11.19%    passed     33.50%\n",
      "  17  CO-AR-6    10.47%    passed     18.25%\n",
      "  18  CO-AR-7    11.13%    passed     27.04%\n",
      "  19  CO-AR-8     8.44%    passed     39.98%\n",
      "  20  CO-AR-9     5.28%    passed     71.71%\n",
      "  21  CO-CH-11   12.83%    passed     26.59%\n",
      "  22  CO-CH-12   12.90%    passed     81.53%\n",
      "  23  CO-CH-18   36.74%    passed     26.33%\n",
      "  24  CO-CH-2     7.56%    passed     77.50%\n",
      "  25  CO-CH-21    6.45%    passed     99.26%\n",
      "  26  CO-CH-23   17.67%    passed     76.93%\n",
      "  27  CO-CH-24   29.98%    passed     76.48%\n",
      "  28  CO-CH-29   40.04%    passed     89.61%\n",
      "  29  CO-CH-39   10.01%    passed     90.78%\n",
      "  30  CO-CH-40    9.92%    passed     89.60%\n",
      "  31  CO-CH-45    7.33%    passed     76.71%\n",
      "  32  CO-CH-46   12.68%    passed     32.10%\n",
      "  33  CO-CH-5     6.65%    passed     98.04%\n",
      "  34  CO-CH-6     9.80%    passed     70.06%\n",
      "  35  CO-CH-8    11.51%    passed     48.53%\n",
      "  36  CO-CH-9    10.39%    passed     54.27%\n",
      "  37  CO-IP-1    10.01%    passed     29.67%\n",
      "  38  CO-IP-10   11.69%    failed      4.93%\n",
      "  39  CO-IP-2    10.04%    passed     39.69%\n",
      "  40  CO-IP-3    11.23%    passed     25.78%\n",
      "  41  CO-IP-4    11.42%    passed     30.03%\n",
      "  42  CO-IP-5    11.42%    passed     18.70%\n",
      "  43  CO-IP-6     7.19%    passed     74.04%\n",
      "  44  CO-IP-7    10.95%    passed     32.40%\n",
      "  45  CO-IP-8     7.20%    passed     91.39%\n",
      "  46  CO-IP-9    11.39%    passed     26.93%\n",
      "  47  CO-MO-1     6.41%    passed     59.37%\n",
      "  48  CO-MO-10   10.88%    passed     62.11%\n",
      "  49  CO-MO-11   13.17%    passed     31.83%\n",
      "  50  CO-MO-13    4.33%    passed     98.25%\n",
      "  51  CO-MO-14    8.31%    passed     40.00%\n",
      "  52  CO-MO-15    8.21%    passed     75.36%\n",
      "  53  CO-MO-16    9.82%    passed     43.58%\n",
      "  54  CO-MO-17   12.04%    passed     29.92%\n",
      "  55  CO-MO-18    8.96%    passed     57.68%\n",
      "  56  CO-MO-19   11.66%    passed     44.96%\n",
      "  57  CO-MO-20    8.82%    passed     64.59%\n",
      "  58  CO-MO-3     6.44%    passed     92.26%\n",
      "  59  CO-MO-4     4.90%    passed     92.91%\n",
      "  60  CO-MO-5    10.46%    passed     45.92%\n",
      "  61  CO-MO-6    12.56%    passed     28.13%\n",
      "  62  CO-MO-7    12.11%    passed      8.93%\n",
      "  63  CO-MO-8     8.57%    passed     85.56%\n",
      "  64  CO-MO-9    12.07%    passed     49.32%\n",
      "  65  CO-PB-1    12.94%    passed     45.54%\n",
      "  66  CO-PB-10   13.45%    passed     57.32%\n",
      "  67  CO-PB-11   12.89%    passed     42.90%\n",
      "  68  CO-PB-12   12.96%    passed     51.15%\n",
      "  69  CO-PB-13   12.59%    passed     42.48%\n",
      "  70  CO-PB-14   13.51%    passed     38.81%\n",
      "  71  CO-PB-15   13.28%    passed     42.53%\n",
      "  72  CO-PB-16   16.97%    passed     81.33%\n",
      "  73  CO-PB-17   13.25%    passed     62.85%\n",
      "  74  CO-PB-18   12.96%    passed     94.08%\n",
      "  75  CO-PB-19   13.18%    passed     91.30%\n",
      "  76  CO-PB-2    12.75%    passed     49.90%\n",
      "  77  CO-PB-20   13.23%    passed     84.27%\n",
      "  78  CO-PB-3    12.16%    passed     59.02%\n",
      "  79  CO-PB-4    14.41%    passed     60.34%\n",
      "  80  CO-PB-5    13.23%    passed     60.46%\n",
      "  81  CO-PB-6    11.60%    passed     99.94%\n",
      "  82  CO-PB-7    11.92%    passed     38.88%\n",
      "  83  CO-PB-8    12.95%    passed     53.79%\n",
      "  84  CO-PB-9    12.54%    passed     53.68%\n",
      "  85  CO-PK-10   12.22%    passed     54.48%\n",
      "  86  CO-PK-11   22.45%    passed     72.43%\n",
      "  87  CO-PK-12   12.08%    passed     57.90%\n",
      "  88  CO-PK-13   37.63%    passed     80.93%\n",
      "  89  CO-PK-14   12.07%    passed     48.61%\n",
      "  90  CO-PK-15   12.29%    passed     52.30%\n",
      "  91  CO-PK-16    9.05%    passed     88.56%\n",
      "  92  CO-PK-17   11.56%    passed     50.82%\n",
      "  93  CO-PK-18    8.21%    passed     83.71%\n",
      "  94  CO-PK-19   18.03%    passed     90.59%\n",
      "  95  CO-PK-2    33.23%    passed     53.03%\n",
      "  96  CO-PK-20   12.05%    passed     52.64%\n",
      "  97  CO-PK-3    12.12%    passed     40.38%\n",
      "  98  CO-PK-4    17.87%    passed     67.77%\n",
      "  99  CO-PK-5    40.74%    passed     22.16%\n",
      " 100  CO-PK-6    11.25%    passed     26.18%\n",
      " 101  CO-PK-9    11.44%    passed     27.01%\n",
      "****  TOTAL      12.41%  1 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.210 seconds\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -205756.029\n",
      "2. Current log-likelihood: -205750.811\n",
      "3. Current log-likelihood: -205747.235\n",
      "4. Current log-likelihood: -205744.086\n",
      "5. Current log-likelihood: -205741.285\n",
      "6. Current log-likelihood: -205738.774\n",
      "7. Current log-likelihood: -205736.512\n",
      "8. Current log-likelihood: -205734.464\n",
      "9. Current log-likelihood: -205732.606\n",
      "10. Current log-likelihood: -205730.914\n",
      "11. Current log-likelihood: -205729.372\n",
      "12. Current log-likelihood: -205727.962\n",
      "13. Current log-likelihood: -205726.670\n",
      "14. Current log-likelihood: -205725.486\n",
      "15. Current log-likelihood: -205724.397\n",
      "Optimal log-likelihood: -205723.399\n",
      "Rate parameters:  A-C: 1.05311  A-G: 2.65416  A-T: 1.78783  C-G: 0.59713  C-T: 2.64366  G-T: 1.00000\n",
      "Base frequencies:  A: 0.248  C: 0.254  G: 0.249  T: 0.248\n",
      "Proportion of invariable sites: 0.011\n",
      "Gamma shape alpha: 0.968\n",
      "Parameters optimization took 15 rounds (6.336 sec)\n",
      "Time for fast ML tree search: 15.452 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 1344 MB RAM!\n",
      "ModelFinder will test up to 484 DNA models (sample size: 19088 epsilon: 0.100) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  GTR+F         210669.631   207 421753.263   421757.824   423379.623\n",
      "  2  GTR+F+I       208996.599   208 418409.197   418413.803   420043.415\n",
      "  3  GTR+F+G4      205710.095   208 411836.189   411840.794   413470.407\n",
      "  4  GTR+F+I+G4    205711.571   209 411841.142   411845.792   413483.217\n",
      "  5  GTR+F+R2      204705.428   209 409828.857   409833.507   411470.931\n",
      "  6  GTR+F+R3      204400.701   211 409223.401   409228.141   410881.189\n",
      "  7  GTR+F+R4      204286.507   213 408999.013   409003.843   410672.515\n",
      "  8  GTR+F+R5      204286.484   215 409002.968   409007.890   410692.184\n",
      " 14  GTR+F+I+R2    204707.790   210 409835.580   409840.274   411485.511\n",
      " 15  GTR+F+I+R3    204400.719   212 409225.437   409230.222   410891.082\n",
      " 16  GTR+F+I+R4    204286.505   214 409001.009   409005.885   410682.368\n",
      " 17  GTR+F+I+R5    204286.483   216 409004.966   409009.934   410702.038\n",
      " 29  SYM+R4        204278.616   210 408977.232   408981.927   410627.163\n",
      " 38  SYM+I+R4      204278.664   211 408979.327   408984.067   410637.115\n",
      " 51  TVM+F+R4      204286.979   212 408997.958   409002.743   410663.603\n",
      " 60  TVM+F+I+R4    204287.038   213 409000.076   409004.906   410673.577\n",
      " 73  TVMe+R4       204278.618   209 408975.236   408979.885   410617.310\n",
      " 82  TVMe+I+R4     204278.683   210 408977.366   408982.061   410627.297\n",
      " 95  TIM3+F+R4     204745.602   211 409913.204   409917.944   411570.992\n",
      "104  TIM3+F+I+R4   204745.650   212 409915.301   409920.085   411580.945\n",
      "117  TIM3e+R4      204733.911   208 409883.822   409888.428   411518.040\n",
      "126  TIM3e+I+R4    204733.963   209 409885.927   409890.576   411528.001\n",
      "139  TIM2+F+R4     204672.222   211 409766.444   409771.184   411424.232\n",
      "148  TIM2+F+I+R4   204672.235   212 409768.470   409773.255   411434.115\n",
      "161  TIM2e+R4      204640.788   208 409697.576   409702.182   411331.794\n",
      "170  TIM2e+I+R4    204640.800   209 409699.600   409704.250   411341.675\n",
      "183  TIM+F+R4      205110.647   211 410643.294   410648.034   412301.082\n",
      "192  TIM+F+I+R4    205110.663   212 410645.326   410650.111   412310.971\n",
      "205  TIMe+R4       205075.534   208 410567.067   410571.673   412201.285\n",
      "214  TIMe+I+R4     205075.554   209 410569.108   410573.758   412211.182\n",
      "227  TPM3u+F+R4    204746.806   210 409913.612   409918.307   411563.543\n",
      "236  TPM3u+F+I+R4  204746.806   211 409915.612   409920.351   411573.400\n",
      "249  TPM3+R4       204734.348   207 409882.697   409887.258   411509.057\n",
      "258  TPM3+I+R4     204734.347   208 409884.693   409889.299   411518.911\n",
      "271  TPM2u+F+R4    204672.442   210 409764.883   409769.578   411414.815\n",
      "280  TPM2u+F+I+R4  204672.439   211 409766.879   409771.618   411424.667\n",
      "293  TPM2+R4       204640.745   207 409695.489   409700.050   411321.850\n",
      "302  TPM2+I+R4     204640.746   208 409697.492   409702.098   411331.710\n",
      "315  K3Pu+F+R4     205111.346   210 410642.692   410647.386   412292.623\n",
      "324  K3Pu+F+I+R4   205111.342   211 410644.684   410649.424   412302.472\n",
      "337  K3P+R4        205075.669   207 410565.338   410569.899   412191.698\n",
      "346  K3P+I+R4      205075.675   208 410567.349   410571.954   412201.567\n",
      "359  TN+F+R4       205128.537   210 410677.073   410681.768   412327.004\n",
      "368  TN+F+I+R4     205128.536   211 410679.071   410683.811   412336.859\n",
      "381  TNe+R4        205093.438   207 410600.876   410605.437   412227.237\n",
      "390  TNe+I+R4      205093.443   208 410602.885   410607.491   412237.103\n",
      "403  HKY+F+R4      205129.277   209 410676.554   410681.204   412318.628\n",
      "412  HKY+F+I+R4    205129.281   210 410678.562   410683.256   412328.493\n",
      "425  K2P+R4        205093.602   206 410599.204   410603.721   412217.708\n",
      "434  K2P+I+R4      205093.606   207 410601.212   410605.773   412227.573\n",
      "447  F81+F+R4      208110.624   208 416637.248   416641.853   418271.465\n",
      "456  F81+F+I+R4    208110.624   209 416639.248   416643.898   418281.323\n",
      "469  JC+R4         208074.816   205 416559.632   416564.105   418170.279\n",
      "478  JC+I+R4       208074.819   206 416561.638   416566.155   418180.142\n",
      "Akaike Information Criterion:           TVMe+R4\n",
      "Corrected Akaike Information Criterion: TVMe+R4\n",
      "Bayesian Information Criterion:         TVMe+R4\n",
      "Best-fit model: TVMe+R4 chosen according to BIC\n",
      "\n",
      "All model information printed to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.model.gz\n",
      "CPU time for ModelFinder: 3020.231 seconds (0h:50m:20s)\n",
      "Wall-clock time for ModelFinder: 314.413 seconds (0h:5m:14s)\n",
      "Generating 10000 samples for ultrafast bootstrap (seed: 990625)...\n",
      "\n",
      "NOTE: 971 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -208074.816\n",
      "2. Current log-likelihood: -204279.738\n",
      "3. Current log-likelihood: -204278.563\n",
      "Optimal log-likelihood: -204278.543\n",
      "Rate parameters:  A-C: 1.06267  A-G: 2.68949  A-T: 1.75069  C-G: 0.61114  C-T: 2.68949  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.807,0.503) (0.157,2.460) (0.035,5.392) (0.001,24.621)\n",
      "Parameters optimization took 3 rounds (2.309 sec)\n",
      "Wrote distance file to... \n",
      "Computing ML distances based on estimated model parameters...\n",
      "Calculating distance matrix: done in 0.0558891 secs using 986.6% CPU\n",
      "Computing ML distances took 0.063431 sec (of wall-clock time) 0.606829 sec (of CPU time)\n",
      "Setting up auxiliary I and S matrices: done in 0.00265192 secs using 172.8% CPU\n",
      "Computing RapidNJ tree took 0.350745 sec (of wall-clock time) 0.327843 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -205598.370\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 26.390 second\n",
      "Computing log-likelihood of 98 initial trees ... 8.944 seconds\n",
      "Current best score: -204245.304\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -204108.355\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 7: -204106.666\n",
      "Iteration 10 / LogL: -204122.313 / Time: 0h:6m:44s\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 11: -204103.653\n",
      "Iteration 20 / LogL: -204204.064 / Time: 0h:7m:32s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -204103.653 / CPU time: 130.109\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 24: -204100.997\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 25: -204099.187\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 28: -204091.744\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 30: -204089.195\n",
      "Iteration 30 / LogL: -204089.195 / Time: 0h:8m:3s (0h:9m:42s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 38: -204080.597\n",
      "Iteration 40 / LogL: -204126.064 / Time: 0h:8m:32s (0h:8m:18s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 48: -204077.337\n",
      "Iteration 50 / LogL: -204096.384 / Time: 0h:9m:1s (0h:7m:34s left)\n",
      "Log-likelihood cutoff on original alignment: -204365.985\n",
      "Iteration 60 / LogL: -204110.358 / Time: 0h:9m:32s (0h:6m:24s left)\n",
      "Iteration 70 / LogL: -204079.821 / Time: 0h:10m:0s (0h:5m:23s left)\n",
      "Iteration 80 / LogL: -204099.447 / Time: 0h:10m:27s (0h:4m:29s left)\n",
      "Iteration 90 / LogL: -204091.008 / Time: 0h:10m:53s (0h:3m:40s left)\n",
      "Iteration 100 / LogL: -204120.098 / Time: 0h:11m:18s (0h:2m:56s left)\n",
      "Log-likelihood cutoff on original alignment: -204365.985\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.997\n",
      "Iteration 110 / LogL: -204106.866 / Time: 0h:11m:46s (0h:5m:23s left)\n",
      "Iteration 120 / LogL: -204275.004 / Time: 0h:12m:10s (0h:4m:39s left)\n",
      "Iteration 130 / LogL: -204077.703 / Time: 0h:12m:38s (0h:4m:0s left)\n",
      "Iteration 140 / LogL: -204089.701 / Time: 0h:13m:3s (0h:3m:22s left)\n",
      "TREE SEARCH COMPLETED AFTER 149 ITERATIONS / Time: 0h:13m:25s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -204077.337\n",
      "2. Current log-likelihood: -204077.272\n",
      "3. Current log-likelihood: -204077.229\n",
      "Optimal log-likelihood: -204077.218\n",
      "Rate parameters:  A-C: 1.06167  A-G: 2.69240  A-T: 1.74888  C-G: 0.61176  C-T: 2.69240  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.807,0.502) (0.156,2.436) (0.036,5.340) (0.001,24.940)\n",
      "Parameters optimization took 3 rounds (1.049 sec)\n",
      "BEST SCORE FOUND : -204077.218\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 10000 replicates...\n",
      "34.187 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex\n",
      "Total tree length: 1.912\n",
      "\n",
      "Total number of iterations: 149\n",
      "CPU time used for tree search: 4423.916 sec (1h:13m:43s)\n",
      "Wall-clock time used for tree search: 483.821 sec (0h:8m:3s)\n",
      "Total CPU time used: 7827.951 sec (2h:10m:27s)\n",
      "Total wall-clock time used: 843.521 sec (0h:14m:3s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex...\n",
      "101 taxa and 458 splits.\n",
      "Consensus tree written to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "Reading input trees file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "Log-likelihood of consensus tree: -204077.202\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.iqtree\n",
      "  Maximum-likelihood tree:       ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.treefile\n",
      "  Likelihood distances:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex\n",
      "  Consensus tree:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "  Screen log file:               ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss40.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.log\n",
      "\n",
      "Date and Time: Mon Aug  4 00:29:30 2025\n",
      "IQ-TREE multicore version 2.4.0 for Linux x86 64-bit built Feb 12 2025\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
      "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
      "\n",
      "Host:    cantley (AVX512, FMA3, 377 GB RAM)\n",
      "Command: iqtree2 -s ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy -m MFP --alrt 10000 -B 10000 -T 10 -redo\n",
      "Seed:    919998 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Aug  4 00:29:30 2025\n",
      "Kernel:  AVX+FMA - 10 threads (10 CPU cores detected)\n",
      "\n",
      "Reading alignment file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy ... Phylip format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 105 sequences with 18208 columns, 18188 distinct patterns\n",
      "8216 parsimony-informative, 4513 singleton sites, 5479 constant sites\n",
      "          Gap/Ambiguity  Composition  p-value\n",
      "Analyzing sequences: done in 0.00179751 secs using 1144% CPU\n",
      "   1  CO-AR-1    11.70%    passed     21.40%\n",
      "   2  CO-AR-10   10.37%    passed     20.37%\n",
      "   3  CO-AR-11   11.09%    passed     29.34%\n",
      "   4  CO-AR-12    6.37%    passed     53.66%\n",
      "   5  CO-AR-13   10.96%    passed     31.20%\n",
      "   6  CO-AR-14    7.19%    passed     69.51%\n",
      "   7  CO-AR-15    7.13%    passed     60.06%\n",
      "   8  CO-AR-16    6.44%    passed     62.18%\n",
      "   9  CO-AR-17   10.55%    passed     44.41%\n",
      "  10  CO-AR-18    7.51%    passed     37.75%\n",
      "  11  CO-AR-19   11.01%    passed     28.04%\n",
      "  12  CO-AR-2    11.27%    passed     17.71%\n",
      "  13  CO-AR-20    7.17%    passed     63.54%\n",
      "  14  CO-AR-3    11.61%    passed     16.55%\n",
      "  15  CO-AR-4    11.10%    passed     69.50%\n",
      "  16  CO-AR-5    11.71%    passed     30.40%\n",
      "  17  CO-AR-6    10.92%    passed     16.50%\n",
      "  18  CO-AR-7    11.64%    passed     24.49%\n",
      "  19  CO-AR-8     8.78%    passed     35.59%\n",
      "  20  CO-AR-9     5.50%    passed     73.52%\n",
      "  21  CO-CH-1    45.26%    passed     32.22%\n",
      "  22  CO-CH-10   60.43%    passed     80.34%\n",
      "  23  CO-CH-11   13.12%    passed     18.28%\n",
      "  24  CO-CH-12   13.35%    passed     70.41%\n",
      "  25  CO-CH-18   38.20%    passed     10.33%\n",
      "  26  CO-CH-2     8.36%    passed     77.06%\n",
      "  27  CO-CH-21    6.75%    passed     97.57%\n",
      "  28  CO-CH-22   46.89%    passed     64.82%\n",
      "  29  CO-CH-23   18.53%    passed     73.86%\n",
      "  30  CO-CH-24   31.07%    passed     89.24%\n",
      "  31  CO-CH-28   61.29%    passed     38.30%\n",
      "  32  CO-CH-29   41.51%    passed     93.71%\n",
      "  33  CO-CH-39   10.41%    passed     83.99%\n",
      "  34  CO-CH-40   10.37%    passed     89.69%\n",
      "  35  CO-CH-45    7.76%    passed     76.32%\n",
      "  36  CO-CH-46   13.17%    passed     31.62%\n",
      "  37  CO-CH-5     6.79%    passed     96.03%\n",
      "  38  CO-CH-6     9.89%    passed     64.99%\n",
      "  39  CO-CH-8    11.82%    passed     48.84%\n",
      "  40  CO-CH-9    10.52%    passed     41.14%\n",
      "  41  CO-IP-1    10.39%    passed     28.80%\n",
      "  42  CO-IP-10   12.21%    failed      4.56%\n",
      "  43  CO-IP-2    10.42%    passed     40.16%\n",
      "  44  CO-IP-3    11.75%    passed     23.05%\n",
      "  45  CO-IP-4    11.95%    passed     26.60%\n",
      "  46  CO-IP-5    11.95%    passed     17.34%\n",
      "  47  CO-IP-6     7.55%    passed     74.00%\n",
      "  48  CO-IP-7    11.45%    passed     34.42%\n",
      "  49  CO-IP-8     7.51%    passed     90.99%\n",
      "  50  CO-IP-9    11.91%    passed     24.88%\n",
      "  51  CO-MO-1     6.77%    passed     58.68%\n",
      "  52  CO-MO-10   11.40%    passed     63.39%\n",
      "  53  CO-MO-11   13.80%    passed     30.54%\n",
      "  54  CO-MO-13    4.59%    passed     96.74%\n",
      "  55  CO-MO-14    8.76%    passed     37.36%\n",
      "  56  CO-MO-15    8.64%    passed     79.93%\n",
      "  57  CO-MO-16   10.32%    passed     43.65%\n",
      "  58  CO-MO-17   12.65%    passed     29.47%\n",
      "  59  CO-MO-18    9.45%    passed     54.56%\n",
      "  60  CO-MO-19   12.12%    passed     47.84%\n",
      "  61  CO-MO-20    9.27%    passed     63.93%\n",
      "  62  CO-MO-3     6.85%    passed     94.37%\n",
      "  63  CO-MO-4     5.20%    passed     92.16%\n",
      "  64  CO-MO-5    10.90%    passed     48.60%\n",
      "  65  CO-MO-6    13.09%    passed     27.70%\n",
      "  66  CO-MO-7    12.66%    passed      8.52%\n",
      "  67  CO-MO-8     8.96%    passed     88.45%\n",
      "  68  CO-MO-9    12.75%    passed     47.83%\n",
      "  69  CO-PB-1    13.51%    passed     40.75%\n",
      "  70  CO-PB-10   14.07%    passed     54.33%\n",
      "  71  CO-PB-11   13.46%    passed     37.74%\n",
      "  72  CO-PB-12   13.52%    passed     49.37%\n",
      "  73  CO-PB-13   13.07%    passed     37.17%\n",
      "  74  CO-PB-14   14.03%    passed     38.41%\n",
      "  75  CO-PB-15   13.83%    passed     37.45%\n",
      "  76  CO-PB-16   17.90%    passed     81.00%\n",
      "  77  CO-PB-17   13.85%    passed     59.66%\n",
      "  78  CO-PB-18   13.52%    passed     92.80%\n",
      "  79  CO-PB-19   13.71%    passed     87.75%\n",
      "  80  CO-PB-2    13.20%    passed     46.51%\n",
      "  81  CO-PB-20   13.77%    passed     79.70%\n",
      "  82  CO-PB-3    12.73%    passed     58.61%\n",
      "  83  CO-PB-4    15.09%    passed     59.67%\n",
      "  84  CO-PB-5    13.79%    passed     52.54%\n",
      "  85  CO-PB-6    12.13%    passed     99.75%\n",
      "  86  CO-PB-7    12.40%    passed     33.88%\n",
      "  87  CO-PB-8    13.52%    passed     51.85%\n",
      "  88  CO-PB-9    13.09%    passed     51.41%\n",
      "  89  CO-PK-10   12.84%    passed     49.29%\n",
      "  90  CO-PK-11   23.10%    passed     72.41%\n",
      "  91  CO-PK-12   12.68%    passed     51.65%\n",
      "  92  CO-PK-13   38.50%    passed     81.88%\n",
      "  93  CO-PK-14   12.66%    passed     43.49%\n",
      "  94  CO-PK-15   12.93%    passed     47.32%\n",
      "  95  CO-PK-16    9.53%    passed     88.31%\n",
      "  96  CO-PK-17   12.15%    passed     51.55%\n",
      "  97  CO-PK-18    8.58%    passed     81.64%\n",
      "  98  CO-PK-19   18.66%    passed     88.97%\n",
      "  99  CO-PK-2    34.16%    passed     54.23%\n",
      " 100  CO-PK-20   12.63%    passed     50.76%\n",
      " 101  CO-PK-3    12.73%    passed     38.63%\n",
      " 102  CO-PK-4    18.79%    passed     68.24%\n",
      " 103  CO-PK-5    41.23%    passed     15.99%\n",
      " 104  CO-PK-6    11.80%    passed     24.31%\n",
      " 105  CO-PK-9    12.01%    passed     26.63%\n",
      "WARNING: 2 sequences contain more than 50% gaps/ambiguity\n",
      "****  TOTAL      14.48%  1 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.256 seconds\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -204788.306\n",
      "2. Current log-likelihood: -204783.861\n",
      "3. Current log-likelihood: -204780.092\n",
      "4. Current log-likelihood: -204776.794\n",
      "5. Current log-likelihood: -204773.878\n",
      "6. Current log-likelihood: -204771.293\n",
      "7. Current log-likelihood: -204768.980\n",
      "8. Current log-likelihood: -204766.905\n",
      "9. Current log-likelihood: -204765.032\n",
      "10. Current log-likelihood: -204763.339\n",
      "11. Current log-likelihood: -204761.802\n",
      "12. Current log-likelihood: -204760.404\n",
      "13. Current log-likelihood: -204759.129\n",
      "14. Current log-likelihood: -204757.964\n",
      "15. Current log-likelihood: -204756.898\n",
      "Optimal log-likelihood: -204755.926\n",
      "Rate parameters:  A-C: 1.05099  A-G: 2.66944  A-T: 1.80756  C-G: 0.59875  C-T: 2.68335  G-T: 1.00000\n",
      "Base frequencies:  A: 0.249  C: 0.254  G: 0.250  T: 0.247\n",
      "Proportion of invariable sites: 0.010\n",
      "Gamma shape alpha: 0.987\n",
      "Parameters optimization took 15 rounds (6.319 sec)\n",
      "Time for fast ML tree search: 14.775 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 1313 MB RAM!\n",
      "ModelFinder will test up to 484 DNA models (sample size: 18208 epsilon: 0.100) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  GTR+F         209730.215   215 419890.429   419895.592   421569.497\n",
      "  2  GTR+F+I       208062.383   216 416556.765   416561.976   418243.642\n",
      "  3  GTR+F+G4      204743.641   216 409919.282   409924.493   411606.159\n",
      "  4  GTR+F+I+G4    204745.022   217 409924.043   409929.302   411618.730\n",
      "  5  GTR+F+R2      203842.239   217 408118.478   408123.737   409813.165\n",
      "  6  GTR+F+R3      203527.724   219 407493.447   407498.804   409203.753\n",
      "  7  GTR+F+R4      203398.857   221 407239.714   407245.170   408965.639\n",
      "+R5 reinitialized from +R4 with factor 0.500\n",
      "+R5 reinitialized from +R4 with factor 0.250\n",
      "+R5 reinitialized from +R4 with factor 0.125\n",
      "+R5 reinitialized from +R4 with factor 0.062\n",
      "+R5 reinitialized from +R4 with factor 0.031\n",
      "  8  GTR+F+R5      203399.315   223 407244.629   407250.184   408986.174\n",
      " 14  GTR+F+I+R2    203841.057   218 408118.113   408123.421   409820.610\n",
      " 15  GTR+F+I+R3    203527.728   220 407495.457   407500.863   409213.572\n",
      " 16  GTR+F+I+R4    203398.862   222 407241.723   407247.229   408975.458\n",
      "+I+R5 reinitialized from +I+R4 with factor 0.500\n",
      "+I+R5 reinitialized from +I+R4 with factor 0.250\n",
      "+I+R5 reinitialized from +I+R4 with factor 0.125\n",
      "+I+R5 reinitialized from +I+R4 with factor 0.062\n",
      "+I+R5 reinitialized from +I+R4 with factor 0.031\n",
      " 17  GTR+F+I+R5    203399.317   224 407246.634   407252.240   408995.988\n",
      " 29  SYM+R4        203387.611   218 407211.223   407216.530   408913.719\n",
      " 38  SYM+I+R4      203387.581   219 407213.163   407218.520   408923.469\n",
      " 51  TVM+F+R4      203398.838   220 407237.676   407243.083   408955.792\n",
      " 60  TVM+F+I+R4    203398.866   221 407239.731   407245.187   408965.657\n",
      " 73  TVMe+R4       203387.515   217 407209.030   407214.289   408903.717\n",
      " 82  TVMe+I+R4     203387.544   218 407211.089   407216.397   408913.585\n",
      " 95  TIM3+F+R4     203866.115   219 408170.231   408175.588   409880.537\n",
      "104  TIM3+F+I+R4   203866.138   220 408172.276   408177.683   409890.392\n",
      "117  TIM3e+R4      203846.821   216 408125.642   408130.853   409812.519\n",
      "126  TIM3e+I+R4    203846.851   217 408127.703   408132.962   409822.390\n",
      "139  TIM2+F+R4     203798.532   219 408035.063   408040.420   409745.369\n",
      "148  TIM2+F+I+R4   203798.529   220 408037.058   408042.464   409755.174\n",
      "161  TIM2e+R4      203755.419   216 407942.839   407948.049   409629.716\n",
      "170  TIM2e+I+R4    203755.414   217 407944.829   407950.088   409639.515\n",
      "183  TIM+F+R4      204242.067   219 408922.133   408927.490   410632.439\n",
      "192  TIM+F+I+R4    204242.065   220 408924.130   408929.536   410642.245\n",
      "205  TIMe+R4       204191.068   216 408814.135   408819.346   410501.012\n",
      "214  TIMe+I+R4     204191.056   217 408816.112   408821.371   410510.798\n",
      "227  TPM3u+F+R4    203866.539   218 408169.079   408174.387   409871.575\n",
      "236  TPM3u+F+I+R4  203866.544   219 408171.088   408176.445   409881.394\n",
      "249  TPM3+R4       203847.039   215 408124.078   408129.240   409803.145\n",
      "258  TPM3+I+R4     203847.029   216 408126.059   408131.269   409812.936\n",
      "271  TPM2u+F+R4    203798.484   218 408032.967   408038.275   409735.464\n",
      "280  TPM2u+F+I+R4  203798.482   219 408034.964   408040.320   409745.270\n",
      "293  TPM2+R4       203755.381   215 407940.762   407945.925   409619.830\n",
      "302  TPM2+I+R4     203755.378   216 407942.757   407947.967   409629.634\n",
      "315  K3Pu+F+R4     204242.225   218 408920.450   408925.758   410622.946\n",
      "324  K3Pu+F+I+R4   204242.207   219 408922.415   408927.772   410632.721\n",
      "337  K3P+R4        204191.078   215 408812.157   408817.319   410491.224\n",
      "346  K3P+I+R4      204191.075   216 408814.151   408819.361   410501.028\n",
      "359  TN+F+R4       204262.571   218 408961.143   408966.451   410663.639\n",
      "368  TN+F+I+R4     204262.572   219 408963.143   408968.500   410673.449\n",
      "381  TNe+R4        204211.586   215 408853.171   408858.334   410532.239\n",
      "390  TNe+I+R4      204211.580   216 408855.160   408860.371   410542.038\n",
      "403  HKY+F+R4      204262.753   217 408959.507   408964.766   410654.194\n",
      "412  HKY+F+I+R4    204262.753   218 408961.506   408966.814   410664.003\n",
      "425  K2P+R4        204211.629   214 408851.258   408856.372   410522.515\n",
      "434  K2P+I+R4      204211.629   215 408853.258   408858.420   410532.325\n",
      "447  F81+F+R4      207311.942   216 415055.883   415061.094   416742.760\n",
      "456  F81+F+I+R4    207311.940   217 415057.880   415063.139   416752.567\n",
      "469  JC+R4         207260.339   213 414946.677   414951.744   416610.126\n",
      "478  JC+I+R4       207260.340   214 414948.680   414953.794   416619.938\n",
      "Akaike Information Criterion:           TVMe+R4\n",
      "Corrected Akaike Information Criterion: TVMe+R4\n",
      "Bayesian Information Criterion:         TVMe+R4\n",
      "Best-fit model: TVMe+R4 chosen according to BIC\n",
      "\n",
      "All model information printed to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.model.gz\n",
      "CPU time for ModelFinder: 3344.641 seconds (0h:55m:44s)\n",
      "Wall-clock time for ModelFinder: 349.616 seconds (0h:5m:49s)\n",
      "Generating 10000 samples for ultrafast bootstrap (seed: 919998)...\n",
      "\n",
      "NOTE: 941 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -207260.339\n",
      "2. Current log-likelihood: -203388.659\n",
      "3. Current log-likelihood: -203387.477\n",
      "Optimal log-likelihood: -203387.450\n",
      "Rate parameters:  A-C: 1.06071  A-G: 2.71631  A-T: 1.76408  C-G: 0.61503  C-T: 2.71631  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.792,0.497) (0.166,2.275) (0.041,5.091) (0.001,24.093)\n",
      "Parameters optimization took 3 rounds (3.837 sec)\n",
      "Wrote distance file to... \n",
      "Computing ML distances based on estimated model parameters...\n",
      "Calculating distance matrix: done in 0.0603617 secs using 999.4% CPU\n",
      "Computing ML distances took 0.067866 sec (of wall-clock time) 0.649092 sec (of CPU time)\n",
      "Setting up auxiliary I and S matrices: done in 0.00656061 secs using 76.24% CPU\n",
      "Computing RapidNJ tree took 0.356226 sec (of wall-clock time) 0.358359 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -204941.140\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 27.089 second\n",
      "Computing log-likelihood of 98 initial trees ... 9.079 seconds\n",
      "Current best score: -203387.450\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -203375.633\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -203372.771\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 3: -203357.688\n",
      "Iteration 10 / LogL: -203450.532 / Time: 0h:7m:29s\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 14: -203352.406\n",
      "Iteration 20 / LogL: -203414.958 / Time: 0h:8m:22s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -203352.406 / CPU time: 143.579\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 28: -203349.175\n",
      "Iteration 30 / LogL: -203390.950 / Time: 0h:8m:50s (0h:10m:11s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 37: -203344.555\n",
      "Iteration 40 / LogL: -203349.063 / Time: 0h:9m:18s (0h:8m:40s left)\n",
      "Iteration 50 / LogL: -203344.974 / Time: 0h:9m:48s (0h:7m:3s left)\n",
      "Log-likelihood cutoff on original alignment: -203590.261\n",
      "Iteration 60 / LogL: -203367.937 / Time: 0h:10m:18s (0h:5m:51s left)\n",
      "Iteration 70 / LogL: -203355.951 / Time: 0h:10m:46s (0h:4m:48s left)\n",
      "Iteration 80 / LogL: -203349.120 / Time: 0h:11m:12s (0h:3m:52s left)\n",
      "Iteration 90 / LogL: -203362.305 / Time: 0h:11m:39s (0h:3m:5s left)\n",
      "Iteration 100 / LogL: -203373.367 / Time: 0h:12m:6s (0h:2m:20s left)\n",
      "Log-likelihood cutoff on original alignment: -203570.461\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.998\n",
      "Iteration 110 / LogL: -203346.789 / Time: 0h:12m:35s (0h:5m:34s left)\n",
      "Iteration 120 / LogL: -203345.003 / Time: 0h:12m:59s (0h:4m:49s left)\n",
      "Iteration 130 / LogL: -203351.622 / Time: 0h:13m:26s (0h:4m:8s left)\n",
      "TREE SEARCH COMPLETED AFTER 138 ITERATIONS / Time: 0h:13m:46s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -203344.555\n",
      "2. Current log-likelihood: -203344.536\n",
      "Optimal log-likelihood: -203344.525\n",
      "Rate parameters:  A-C: 1.05903  A-G: 2.71793  A-T: 1.76049  C-G: 0.61546  C-T: 2.71793  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.793,0.497) (0.165,2.279) (0.041,5.089) (0.001,24.548)\n",
      "Parameters optimization took 2 rounds (0.776 sec)\n",
      "BEST SCORE FOUND : -203344.525\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 10000 replicates...\n",
      "34.150 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex\n",
      "Total tree length: 2.065\n",
      "\n",
      "Total number of iterations: 138\n",
      "CPU time used for tree search: 4258.009 sec (1h:10m:58s)\n",
      "Wall-clock time used for tree search: 467.670 sec (0h:7m:47s)\n",
      "Total CPU time used: 7996.626 sec (2h:13m:16s)\n",
      "Total wall-clock time used: 863.719 sec (0h:14m:23s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex...\n",
      "105 taxa and 477 splits.\n",
      "Consensus tree written to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "Reading input trees file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "Log-likelihood of consensus tree: -203344.526\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.iqtree\n",
      "  Maximum-likelihood tree:       ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.treefile\n",
      "  Likelihood distances:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.splits.nex\n",
      "  Consensus tree:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.contree\n",
      "  Screen log file:               ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.min4.phy.log\n",
      "\n",
      "Date and Time: Mon Aug  4 00:43:55 2025\n",
      "IQ-TREE multicore version 2.4.0 for Linux x86 64-bit built Feb 12 2025\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
      "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
      "\n",
      "Host:    cantley (AVX512, FMA3, 377 GB RAM)\n",
      "Command: iqtree2 -s ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy -m MFP --alrt 10000 -B 10000 -T 10 -redo\n",
      "Seed:    21046 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Aug  4 00:43:56 2025\n",
      "Kernel:  AVX+FMA - 10 threads (10 CPU cores detected)\n",
      "\n",
      "Reading alignment file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy ... Phylip format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 71 sequences with 18208 columns, 17172 distinct patterns\n",
      "6694 parsimony-informative, 3492 singleton sites, 8022 constant sites\n",
      "          Gap/Ambiguity  Composition  p-value\n",
      "Analyzing sequences: done in 0.00458328 secs using 796.4% CPU\n",
      "   1  CO-CH-1    45.26%    passed     29.23%\n",
      "   2  CO-CH-2     8.36%    passed     78.33%\n",
      "   3  CO-CH-21    6.75%    passed     96.28%\n",
      "   4  CO-CH-22   46.89%    passed     59.26%\n",
      "   5  CO-CH-23   18.53%    passed     68.45%\n",
      "   6  CO-CH-24   31.07%    passed     86.76%\n",
      "   7  CO-IP-1    10.39%    passed     23.59%\n",
      "   8  CO-IP-10   12.21%    failed      3.54%\n",
      "   9  CO-IP-2    10.42%    passed     34.29%\n",
      "  10  CO-IP-3    11.75%    passed     18.66%\n",
      "  11  CO-IP-4    11.95%    passed     21.72%\n",
      "  12  CO-IP-5    11.95%    passed     13.94%\n",
      "  13  CO-IP-6     7.55%    passed     66.61%\n",
      "  14  CO-IP-7    11.45%    passed     30.11%\n",
      "  15  CO-IP-8     7.51%    passed     85.84%\n",
      "  16  CO-IP-9    11.91%    passed     20.48%\n",
      "  17  CO-MO-1     6.77%    passed     58.83%\n",
      "  18  CO-MO-10   11.40%    passed     60.27%\n",
      "  19  CO-MO-11   13.80%    passed     26.89%\n",
      "  20  CO-MO-13    4.59%    passed     98.01%\n",
      "  21  CO-MO-14    8.76%    passed     35.77%\n",
      "  22  CO-MO-15    8.64%    passed     73.71%\n",
      "  23  CO-MO-16   10.32%    passed     41.21%\n",
      "  24  CO-MO-17   12.65%    passed     25.91%\n",
      "  25  CO-MO-18    9.45%    passed     50.15%\n",
      "  26  CO-MO-19   12.12%    passed     43.03%\n",
      "  27  CO-MO-20    9.27%    passed     60.04%\n",
      "  28  CO-MO-3     6.85%    passed     95.63%\n",
      "  29  CO-MO-4     5.20%    passed     92.20%\n",
      "  30  CO-MO-5    10.90%    passed     44.11%\n",
      "  31  CO-MO-6    13.09%    passed     25.14%\n",
      "  32  CO-MO-7    12.66%    passed      7.42%\n",
      "  33  CO-MO-8     8.96%    passed     84.74%\n",
      "  34  CO-MO-9    12.75%    passed     45.59%\n",
      "  35  CO-PB-1    13.51%    passed     35.54%\n",
      "  36  CO-PB-10   14.07%    passed     50.26%\n",
      "  37  CO-PB-11   13.46%    passed     33.56%\n",
      "  38  CO-PB-12   13.52%    passed     44.14%\n",
      "  39  CO-PB-13   13.07%    passed     33.93%\n",
      "  40  CO-PB-14   14.03%    passed     32.96%\n",
      "  41  CO-PB-15   13.83%    passed     31.68%\n",
      "  42  CO-PB-16   17.90%    passed     75.54%\n",
      "  43  CO-PB-17   13.85%    passed     55.00%\n",
      "  44  CO-PB-18   13.52%    passed     88.74%\n",
      "  45  CO-PB-19   13.71%    passed     82.41%\n",
      "  46  CO-PB-2    13.20%    passed     41.26%\n",
      "  47  CO-PB-20   13.77%    passed     73.56%\n",
      "  48  CO-PB-3    12.73%    passed     55.12%\n",
      "  49  CO-PB-4    15.09%    passed     57.24%\n",
      "  50  CO-PB-5    13.79%    passed     46.57%\n",
      "  51  CO-PB-6    12.13%    passed     99.86%\n",
      "  52  CO-PB-7    12.40%    passed     29.48%\n",
      "  53  CO-PB-8    13.52%    passed     46.29%\n",
      "  54  CO-PB-9    13.09%    passed     48.40%\n",
      "  55  CO-PK-10   12.84%    passed     46.84%\n",
      "  56  CO-PK-11   23.10%    passed     69.67%\n",
      "  57  CO-PK-12   12.68%    passed     49.71%\n",
      "  58  CO-PK-13   38.50%    passed     85.41%\n",
      "  59  CO-PK-14   12.66%    passed     41.36%\n",
      "  60  CO-PK-15   12.93%    passed     46.16%\n",
      "  61  CO-PK-16    9.53%    passed     89.82%\n",
      "  62  CO-PK-17   12.15%    passed     49.91%\n",
      "  63  CO-PK-18    8.58%    passed     78.36%\n",
      "  64  CO-PK-19   18.66%    passed     92.07%\n",
      "  65  CO-PK-2    34.16%    passed     54.55%\n",
      "  66  CO-PK-20   12.63%    passed     44.59%\n",
      "  67  CO-PK-3    12.73%    passed     34.88%\n",
      "  68  CO-PK-4    18.79%    passed     70.36%\n",
      "  69  CO-PK-5    41.23%    passed     19.82%\n",
      "  70  CO-PK-6    11.80%    passed     23.28%\n",
      "  71  CO-PK-9    12.01%    passed     24.93%\n",
      "****  TOTAL      14.38%  1 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.170 seconds\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -155496.484\n",
      "2. Current log-likelihood: -155492.815\n",
      "3. Current log-likelihood: -155489.628\n",
      "4. Current log-likelihood: -155486.779\n",
      "5. Current log-likelihood: -155484.245\n",
      "6. Current log-likelihood: -155481.997\n",
      "7. Current log-likelihood: -155479.983\n",
      "8. Current log-likelihood: -155478.177\n",
      "9. Current log-likelihood: -155476.545\n",
      "10. Current log-likelihood: -155475.069\n",
      "11. Current log-likelihood: -155473.724\n",
      "12. Current log-likelihood: -155472.498\n",
      "13. Current log-likelihood: -155471.374\n",
      "14. Current log-likelihood: -155470.343\n",
      "Optimal log-likelihood: -155469.350\n",
      "Rate parameters:  A-C: 1.06212  A-G: 2.67720  A-T: 1.82118  C-G: 0.61670  C-T: 2.71134  G-T: 1.00000\n",
      "Base frequencies:  A: 0.249  C: 0.254  G: 0.250  T: 0.247\n",
      "Proportion of invariable sites: 0.111\n",
      "Gamma shape alpha: 0.939\n",
      "Parameters optimization took 14 rounds (3.859 sec)\n",
      "Time for fast ML tree search: 7.680 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 1050 MB RAM!\n",
      "ModelFinder will test up to 484 DNA models (sample size: 18208 epsilon: 0.100) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  GTR+F         159640.989   147 319575.978   319578.388   320723.992\n",
      "  2  GTR+F+I       157086.365   148 314468.730   314471.172   315624.553\n",
      "  3  GTR+F+G4      155452.180   148 311200.360   311202.802   312356.183\n",
      "  4  GTR+F+I+G4    155455.609   149 311209.217   311211.693   312372.850\n",
      "  5  GTR+F+R2      155176.363   149 310650.726   310653.201   311814.359\n",
      "  6  GTR+F+R3      154984.694   151 310271.388   310273.930   311450.640\n",
      "  7  GTR+F+R4      154913.241   153 310132.482   310135.093   311327.354\n",
      "  8  GTR+F+R5      154912.768   155 310135.537   310138.215   311346.027\n",
      " 14  GTR+F+I+R2    155092.783   150 310485.565   310488.074   311657.008\n",
      " 15  GTR+F+I+R3    154948.116   152 310200.231   310202.807   311387.293\n",
      " 16  GTR+F+I+R4    154913.797   154 310135.595   310138.239   311338.276\n",
      " 17  GTR+F+I+R5    154913.774   156 310139.548   310142.262   311357.848\n",
      " 29  SYM+R4        154908.067   150 310116.133   310118.642   311287.576\n",
      " 51  TVM+F+R4      154912.925   152 310129.851   310132.427   311316.913\n",
      " 73  TVMe+R4       154907.752   149 310113.504   310115.979   311277.137\n",
      " 95  TIM3+F+R4     155263.381   151 310828.762   310831.304   312008.014\n",
      "117  TIM3e+R4      155254.670   148 310805.341   310807.783   311961.164\n",
      "139  TIM2+F+R4     155200.322   151 310702.643   310705.185   311881.895\n",
      "161  TIM2e+R4      155171.566   148 310639.131   310641.573   311794.954\n",
      "183  TIM+F+R4      155530.370   151 311362.740   311365.283   312541.992\n",
      "205  TIMe+R4       155498.059   148 311292.118   311294.561   312447.942\n",
      "227  TPM3u+F+R4    155262.940   150 310825.880   310828.389   311997.323\n",
      "249  TPM3+R4       155254.155   147 310802.310   310804.720   311950.324\n",
      "271  TPM2u+F+R4    155199.875   150 310699.750   310702.258   311871.192\n",
      "293  TPM2+R4       155171.262   147 310636.525   310638.934   311784.538\n",
      "315  K3Pu+F+R4     155530.014   150 311360.028   311362.536   312531.470\n",
      "337  K3P+R4        155497.763   147 311289.526   311291.935   312437.539\n",
      "359  TN+F+R4       155547.885   150 311395.771   311398.280   312567.213\n",
      "381  TNe+R4        155515.646   147 311325.291   311327.701   312473.305\n",
      "403  HKY+F+R4      155547.856   149 311393.712   311396.188   312557.345\n",
      "425  K2P+R4        155515.595   146 311323.190   311325.566   312463.394\n",
      "447  F81+F+R4      157732.645   148 315761.290   315763.732   316917.113\n",
      "469  JC+R4         157700.062   145 315690.124   315692.468   316822.518\n",
      "Akaike Information Criterion:           TVMe+R4\n",
      "Corrected Akaike Information Criterion: TVMe+R4\n",
      "Bayesian Information Criterion:         TVMe+R4\n",
      "Best-fit model: TVMe+R4 chosen according to BIC\n",
      "\n",
      "All model information printed to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.model.gz\n",
      "CPU time for ModelFinder: 1564.001 seconds (0h:26m:4s)\n",
      "Wall-clock time for ModelFinder: 162.428 seconds (0h:2m:42s)\n",
      "Generating 10000 samples for ultrafast bootstrap (seed: 21046)...\n",
      "\n",
      "NOTE: 813 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -157700.062\n",
      "2. Current log-likelihood: -154907.857\n",
      "3. Current log-likelihood: -154906.997\n",
      "Optimal log-likelihood: -154906.989\n",
      "Rate parameters:  A-C: 1.07708  A-G: 2.73232  A-T: 1.80236  C-G: 0.62560  C-T: 2.73232  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.683,0.358) (0.243,1.684) (0.073,4.472) (0.001,19.865)\n",
      "Parameters optimization took 3 rounds (2.234 sec)\n",
      "Wrote distance file to... \n",
      "Computing ML distances based on estimated model parameters...\n",
      "Calculating distance matrix: done in 0.0271439 secs using 995% CPU\n",
      "Computing ML distances took 0.034733 sec (of wall-clock time) 0.319368 sec (of CPU time)\n",
      "Setting up auxiliary I and S matrices: done in 0.00135771 secs using 207.5% CPU\n",
      "Computing RapidNJ tree took 0.236717 sec (of wall-clock time) 0.213957 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -155665.432\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 14.456 second\n",
      "Computing log-likelihood of 98 initial trees ... 7.705 seconds\n",
      "Current best score: -154906.989\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -154900.640\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -154807.734\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 4: -154805.638\n",
      "Iteration 10 / LogL: -154807.764 / Time: 0h:3m:43s\n",
      "Iteration 20 / LogL: -154805.695 / Time: 0h:4m:16s\n",
      "Finish initializing candidate tree set (5)\n",
      "Current best tree score: -154805.638 / CPU time: 87.360\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 30 / LogL: -154806.501 / Time: 0h:4m:35s (0h:4m:48s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "UPDATE BEST LOG-LIKELIHOOD: -154805.351\n",
      "Iteration 40 / LogL: -154826.685 / Time: 0h:4m:53s (0h:3m:34s left)\n",
      "Iteration 50 / LogL: -154805.657 / Time: 0h:5m:10s (0h:2m:43s left)\n",
      "Log-likelihood cutoff on original alignment: -154986.747\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 56: -154805.107\n",
      "Iteration 60 / LogL: -154805.509 / Time: 0h:5m:30s (0h:4m:33s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "UPDATE BEST LOG-LIKELIHOOD: -154804.892\n",
      "Iteration 70 / LogL: -154818.547 / Time: 0h:5m:48s (0h:3m:51s left)\n",
      "Iteration 80 / LogL: -154805.253 / Time: 0h:6m:3s (0h:3m:13s left)\n",
      "Iteration 90 / LogL: -154805.515 / Time: 0h:6m:23s (0h:2m:43s left)\n",
      "Iteration 100 / LogL: -154963.171 / Time: 0h:6m:39s (0h:2m:14s left)\n",
      "Log-likelihood cutoff on original alignment: -154986.747\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.999\n",
      "Iteration 110 / LogL: -154806.359 / Time: 0h:6m:59s (0h:3m:32s left)\n",
      "Iteration 120 / LogL: -154809.622 / Time: 0h:7m:15s (0h:3m:3s left)\n",
      "Iteration 130 / LogL: -154804.894 / Time: 0h:7m:33s (0h:2m:37s left)\n",
      "Iteration 140 / LogL: -154842.611 / Time: 0h:7m:50s (0h:2m:13s left)\n",
      "Iteration 150 / LogL: -154806.483 / Time: 0h:8m:9s (0h:1m:49s left)\n",
      "Log-likelihood cutoff on original alignment: -154986.804\n",
      "TREE SEARCH COMPLETED AFTER 157 ITERATIONS / Time: 0h:8m:24s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -154804.892\n",
      "2. Current log-likelihood: -154804.871\n",
      "3. Current log-likelihood: -154804.854\n",
      "4. Current log-likelihood: -154804.834\n",
      "5. Current log-likelihood: -154804.818\n",
      "6. Current log-likelihood: -154804.802\n",
      "7. Current log-likelihood: -154804.786\n",
      "Optimal log-likelihood: -154804.775\n",
      "Rate parameters:  A-C: 1.08240  A-G: 2.73341  A-T: 1.80457  C-G: 0.62103  C-T: 2.73341  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Site proportion and rates:  (0.696,0.365) (0.233,1.749) (0.069,4.566) (0.001,19.621)\n",
      "Parameters optimization took 7 rounds (4.545 sec)\n",
      "BEST SCORE FOUND : -154804.775\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 10000 replicates...\n",
      "22.555 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.splits.nex\n",
      "Total tree length: 1.493\n",
      "\n",
      "Total number of iterations: 157\n",
      "CPU time used for tree search: 3067.455 sec (0h:51m:7s)\n",
      "Wall-clock time used for tree search: 335.600 sec (0h:5m:35s)\n",
      "Total CPU time used: 4929.252 sec (1h:22m:9s)\n",
      "Total wall-clock time used: 533.661 sec (0h:8m:53s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.splits.nex...\n",
      "71 taxa and 276 splits.\n",
      "Consensus tree written to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.contree\n",
      "Reading input trees file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.contree\n",
      "Log-likelihood of consensus tree: -154806.655\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.iqtree\n",
      "  Maximum-likelihood tree:       ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.treefile\n",
      "  Likelihood distances:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.splits.nex\n",
      "  Consensus tree:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.contree\n",
      "  Screen log file:               ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-AR.no-CH-except-OA.min4.phy.log\n",
      "\n",
      "Date and Time: Mon Aug  4 00:52:50 2025\n",
      "IQ-TREE multicore version 2.4.0 for Linux x86 64-bit built Feb 12 2025\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
      "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
      "\n",
      "Host:    cantley (AVX512, FMA3, 377 GB RAM)\n",
      "Command: iqtree2 -s ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy -m MFP --alrt 10000 -B 10000 -T 10 -redo\n",
      "Seed:    535148 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Aug  4 00:52:50 2025\n",
      "Kernel:  AVX+FMA - 10 threads (10 CPU cores detected)\n",
      "\n",
      "Reading alignment file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy ... Phylip format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 65 sequences with 18208 columns, 15473 distinct patterns\n",
      "5811 parsimony-informative, 2390 singleton sites, 10007 constant sites\n",
      "          Gap/Ambiguity  Composition  p-value\n",
      "Analyzing sequences: done in 0.00110324 secs using 1278% CPU\n",
      "   1  CO-IP-1    10.39%    passed     22.78%\n",
      "   2  CO-IP-10   12.21%    failed      3.42%\n",
      "   3  CO-IP-2    10.42%    passed     33.33%\n",
      "   4  CO-IP-3    11.75%    passed     18.02%\n",
      "   5  CO-IP-4    11.95%    passed     21.00%\n",
      "   6  CO-IP-5    11.95%    passed     13.46%\n",
      "   7  CO-IP-6     7.55%    passed     65.26%\n",
      "   8  CO-IP-7    11.45%    passed     29.70%\n",
      "   9  CO-IP-8     7.51%    passed     84.80%\n",
      "  10  CO-IP-9    11.91%    passed     19.85%\n",
      "  11  CO-MO-1     6.77%    passed     59.59%\n",
      "  12  CO-MO-10   11.40%    passed     60.24%\n",
      "  13  CO-MO-11   13.80%    passed     26.62%\n",
      "  14  CO-MO-13    4.59%    passed     98.12%\n",
      "  15  CO-MO-14    8.76%    passed     36.11%\n",
      "  16  CO-MO-15    8.64%    passed     72.66%\n",
      "  17  CO-MO-16   10.32%    passed     41.30%\n",
      "  18  CO-MO-17   12.65%    passed     25.54%\n",
      "  19  CO-MO-18    9.45%    passed     49.87%\n",
      "  20  CO-MO-19   12.12%    passed     42.48%\n",
      "  21  CO-MO-20    9.27%    passed     59.74%\n",
      "  22  CO-MO-3     6.85%    passed     96.09%\n",
      "  23  CO-MO-4     5.20%    passed     92.55%\n",
      "  24  CO-MO-5    10.90%    passed     43.66%\n",
      "  25  CO-MO-6    13.09%    passed     25.03%\n",
      "  26  CO-MO-7    12.66%    passed      7.34%\n",
      "  27  CO-MO-8     8.96%    passed     84.09%\n",
      "  28  CO-MO-9    12.75%    passed     45.71%\n",
      "  29  CO-PB-1    13.51%    passed     34.88%\n",
      "  30  CO-PB-10   14.07%    passed     49.43%\n",
      "  31  CO-PB-11   13.46%    passed     32.98%\n",
      "  32  CO-PB-12   13.52%    passed     43.25%\n",
      "  33  CO-PB-13   13.07%    passed     33.78%\n",
      "  34  CO-PB-14   14.03%    passed     32.20%\n",
      "  35  CO-PB-15   13.83%    passed     30.62%\n",
      "  36  CO-PB-16   17.90%    passed     74.39%\n",
      "  37  CO-PB-17   13.85%    passed     54.52%\n",
      "  38  CO-PB-18   13.52%    passed     88.00%\n",
      "  39  CO-PB-19   13.71%    passed     81.34%\n",
      "  40  CO-PB-2    13.20%    passed     40.22%\n",
      "  41  CO-PB-20   13.77%    passed     72.42%\n",
      "  42  CO-PB-3    12.73%    passed     54.36%\n",
      "  43  CO-PB-4    15.09%    passed     57.17%\n",
      "  44  CO-PB-5    13.79%    passed     45.54%\n",
      "  45  CO-PB-6    12.13%    passed     99.86%\n",
      "  46  CO-PB-7    12.40%    passed     28.95%\n",
      "  47  CO-PB-8    13.52%    passed     45.53%\n",
      "  48  CO-PB-9    13.09%    passed     48.24%\n",
      "  49  CO-PK-10   12.84%    passed     46.76%\n",
      "  50  CO-PK-11   23.10%    passed     69.03%\n",
      "  51  CO-PK-12   12.68%    passed     49.69%\n",
      "  52  CO-PK-13   38.50%    passed     85.63%\n",
      "  53  CO-PK-14   12.66%    passed     41.35%\n",
      "  54  CO-PK-15   12.93%    passed     46.37%\n",
      "  55  CO-PK-16    9.53%    passed     90.49%\n",
      "  56  CO-PK-17   12.15%    passed     50.28%\n",
      "  57  CO-PK-18    8.58%    passed     78.08%\n",
      "  58  CO-PK-19   18.66%    passed     92.66%\n",
      "  59  CO-PK-2    34.16%    passed     55.13%\n",
      "  60  CO-PK-20   12.63%    passed     43.75%\n",
      "  61  CO-PK-3    12.73%    passed     34.67%\n",
      "  62  CO-PK-4    18.79%    passed     71.32%\n",
      "  63  CO-PK-5    41.23%    passed     20.59%\n",
      "  64  CO-PK-6    11.80%    passed     23.46%\n",
      "  65  CO-PK-9    12.01%    passed     25.02%\n",
      "****  TOTAL      13.30%  1 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.086 seconds\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -134662.413\n",
      "Optimal log-likelihood: -134662.357\n",
      "Rate parameters:  A-C: 1.05350  A-G: 2.62836  A-T: 1.79352  C-G: 0.61318  C-T: 2.67976  G-T: 1.00000\n",
      "Base frequencies:  A: 0.249  C: 0.255  G: 0.250  T: 0.247\n",
      "Proportion of invariable sites: 0.275\n",
      "Gamma shape alpha: 0.965\n",
      "Parameters optimization took 1 rounds (0.175 sec)\n",
      "Time for fast ML tree search: 1.327 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 916 MB RAM!\n",
      "ModelFinder will test up to 484 DNA models (sample size: 18208 epsilon: 0.100) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  GTR+F         139664.665   135 279599.329   279601.361   280653.627\n",
      "  2  GTR+F+I       135489.163   136 271250.326   271252.388   272312.434\n",
      "  3  GTR+F+G4      134733.747   136 269739.494   269741.556   270801.602\n",
      "  4  GTR+F+I+G4    134662.429   137 269598.858   269600.950   270668.775\n",
      "  5  GTR+F+R2      134847.061   137 269968.122   269970.214   271038.039\n",
      "  6  GTR+F+R3      134527.904   139 269333.809   269335.963   270419.345\n",
      "  7  GTR+F+R4      134442.219   141 269166.437   269168.654   270267.593\n",
      "  8  GTR+F+R5      134440.102   143 269166.204   269168.484   270282.979\n",
      " 14  GTR+F+I+R2    134545.286   138 269366.573   269368.696   270444.300\n",
      " 15  GTR+F+I+R3    134436.576   140 269153.153   269155.338   270246.499\n",
      " 16  GTR+F+I+R4    134433.892   142 269151.784   269154.032   270260.749\n",
      " 37  SYM+I+R3      134434.550   137 269143.099   269145.192   270213.017\n",
      " 59  TVM+F+I+R3    134436.491   139 269150.981   269153.135   270236.518\n",
      " 81  TVMe+I+R3     134434.652   136 269141.305   269143.367   270203.413\n",
      "103  TIM3+F+I+R3   134725.300   138 269726.601   269728.724   270804.328\n",
      "125  TIM3e+I+R3    134720.768   135 269711.535   269713.567   270765.834\n",
      "147  TIM2+F+I+R3   134682.912   138 269641.825   269643.948   270719.552\n",
      "169  TIM2e+I+R3    134660.873   135 269591.745   269593.777   270646.043\n",
      "191  TIM+F+I+R3    134956.083   138 270188.166   270190.289   271265.893\n",
      "213  TIMe+I+R3     134931.446   135 270132.892   270134.924   271187.190\n",
      "235  TPM3u+F+I+R3  134725.306   137 269724.612   269726.704   270794.529\n",
      "257  TPM3+I+R3     134720.833   134 269709.667   269711.669   270756.155\n",
      "279  TPM2u+F+I+R3  134683.020   137 269640.040   269642.133   270709.957\n",
      "301  TPM2+I+R3     134661.208   134 269590.417   269592.419   270636.905\n",
      "323  K3Pu+F+I+R3   134956.129   137 270186.259   270188.351   271256.176\n",
      "345  K3P+I+R3      134931.659   134 270131.318   270133.319   271177.806\n",
      "367  TN+F+I+R3     134970.043   137 270214.085   270216.178   271284.003\n",
      "389  TNe+I+R3      134945.400   134 270158.801   270160.803   271205.289\n",
      "411  HKY+F+I+R3    134970.096   136 270212.193   270214.255   271274.301\n",
      "433  K2P+I+R3      134945.619   133 270157.238   270159.210   271195.917\n",
      "455  F81+F+I+R3    136744.155   135 273758.309   273760.341   274812.608\n",
      "477  JC+I+R3       136719.301   132 273702.602   273704.544   274733.471\n",
      "Akaike Information Criterion:           TVMe+I+R3\n",
      "Corrected Akaike Information Criterion: TVMe+I+R3\n",
      "Bayesian Information Criterion:         TVMe+I+R3\n",
      "Best-fit model: TVMe+I+R3 chosen according to BIC\n",
      "\n",
      "All model information printed to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.model.gz\n",
      "CPU time for ModelFinder: 1175.431 seconds (0h:19m:35s)\n",
      "Wall-clock time for ModelFinder: 120.964 seconds (0h:2m:0s)\n",
      "Generating 10000 samples for ultrafast bootstrap (seed: 535148)...\n",
      "\n",
      "NOTE: 688 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -136719.301\n",
      "2. Current log-likelihood: -134435.380\n",
      "3. Current log-likelihood: -134434.668\n",
      "Optimal log-likelihood: -134434.649\n",
      "Rate parameters:  A-C: 1.06485  A-G: 2.67741  A-T: 1.78512  C-G: 0.62022  C-T: 2.67741  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Proportion of invariable sites: 0.405\n",
      "Site proportion and rates:  (0.478,1.059) (0.116,4.058) (0.001,20.616)\n",
      "Parameters optimization took 3 rounds (0.896 sec)\n",
      "Wrote distance file to... \n",
      "Computing ML distances based on estimated model parameters...\n",
      "Calculating distance matrix: done in 0.0213775 secs using 970.3% CPU\n",
      "Computing ML distances took 0.031161 sec (of wall-clock time) 0.281618 sec (of CPU time)\n",
      "Setting up auxiliary I and S matrices: done in 0.00262128 secs using 161.8% CPU\n",
      "Computing RapidNJ tree took 0.219799 sec (of wall-clock time) 0.196381 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -135035.495\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 10.979 second\n",
      "Computing log-likelihood of 98 initial trees ... 4.692 seconds\n",
      "Current best score: -134332.932\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -134246.609\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -134246.068\n",
      "Iteration 10 / LogL: -134246.093 / Time: 0h:2m:44s\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 18: -134242.384\n",
      "Iteration 20 / LogL: -134246.181 / Time: 0h:3m:8s\n",
      "Finish initializing candidate tree set (7)\n",
      "Current best tree score: -134242.384 / CPU time: 61.963\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 30 / LogL: -134317.929 / Time: 0h:3m:19s (0h:3m:59s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 32: -134241.506\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 40: -134241.242\n",
      "Iteration 40 / LogL: -134241.242 / Time: 0h:3m:33s (0h:3m:57s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 48: -134239.730\n",
      "Iteration 50 / LogL: -134241.653 / Time: 0h:3m:46s (0h:3m:31s left)\n",
      "Log-likelihood cutoff on original alignment: -134384.201\n",
      "Iteration 60 / LogL: -134242.380 / Time: 0h:4m:1s (0h:3m:0s left)\n",
      "Iteration 70 / LogL: -134243.288 / Time: 0h:4m:15s (0h:2m:31s left)\n",
      "Iteration 80 / LogL: -134241.628 / Time: 0h:4m:27s (0h:2m:6s left)\n",
      "Iteration 90 / LogL: -134248.127 / Time: 0h:4m:38s (0h:1m:42s left)\n",
      "Iteration 100 / LogL: -134239.795 / Time: 0h:4m:52s (0h:1m:23s left)\n",
      "Log-likelihood cutoff on original alignment: -134384.201\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.998\n",
      "Iteration 110 / LogL: -134241.195 / Time: 0h:5m:5s (0h:2m:32s left)\n",
      "Iteration 120 / LogL: -134246.419 / Time: 0h:5m:17s (0h:2m:11s left)\n",
      "Iteration 130 / LogL: -134241.547 / Time: 0h:5m:29s (0h:1m:53s left)\n",
      "Iteration 140 / LogL: -134240.955 / Time: 0h:5m:43s (0h:1m:35s left)\n",
      "TREE SEARCH COMPLETED AFTER 149 ITERATIONS / Time: 0h:5m:55s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -134239.730\n",
      "Optimal log-likelihood: -134239.719\n",
      "Rate parameters:  A-C: 1.07556  A-G: 2.68942  A-T: 1.79447  C-G: 0.61827  C-T: 2.68942  G-T: 1.00000\n",
      "Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250\n",
      "Proportion of invariable sites: 0.400\n",
      "Site proportion and rates:  (0.478,1.033) (0.120,4.012) (0.001,19.872)\n",
      "Parameters optimization took 1 rounds (0.376 sec)\n",
      "BEST SCORE FOUND : -134239.719\n",
      "\n",
      "Testing tree branches by SH-like aLRT with 10000 replicates...\n",
      "20.396 sec.\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.splits.nex\n",
      "Total tree length: 1.209\n",
      "\n",
      "Total number of iterations: 149\n",
      "CPU time used for tree search: 2086.678 sec (0h:34m:46s)\n",
      "Wall-clock time used for tree search: 229.514 sec (0h:3m:49s)\n",
      "Total CPU time used: 3483.165 sec (0h:58m:3s)\n",
      "Total wall-clock time used: 378.015 sec (0h:6m:18s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.splits.nex...\n",
      "65 taxa and 262 splits.\n",
      "Consensus tree written to ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.contree\n",
      "Reading input trees file ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.contree\n",
      "Log-likelihood of consensus tree: -134248.408\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.iqtree\n",
      "  Maximum-likelihood tree:       ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.treefile\n",
      "  Likelihood distances:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.splits.nex\n",
      "  Consensus tree:                ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.contree\n",
      "  Screen log file:               ../05_vcf2phylip/chenopodium_20250304b-tetraploid.indivmiss60.locimiss50.thin10.maf0.01.LD0.8-1000.no-CH-AR.min4.phy.log\n",
      "\n",
      "Date and Time: Mon Aug  4 00:59:09 2025\n"
     ]
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "# should clean this up so that it makes subdirectories?\n",
    "\n",
    "mkdir ../06_phylos/analysis-iqtree\n",
    "cd ../analysis-iqtree\n",
    "for file in ../05_vcf2phylip/*.phy; do \n",
    "    iqtree2 -s $file -m MFP --alrt 10000 -B 10000 -T 10 -redo\n",
    "cd ../06_phylos/\n",
    "done\n",
    "\n",
    "# there's a more elegant way to do this but I'm not sure what it is\n",
    "EXTENSIONS=('contree' 'iqtree' 'treefile' 'mldist' 'nex' 'log' 'bionj' 'ckp.gz' 'model.gz')\n",
    "for EXT in \"${EXTENSIONS[@]}\"; do \n",
    "    mv ../05_vcf2phylip/*.$EXT ../06_phylos/analysis-iqtree/\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
